vendor/sigs.k8s.io/json/internal/golang/encoding/json/kubernetes_patch.go

```diff
@@ -69,6 +71,14 @@ func (d *Decoder) DisallowDuplicateFields() {
 	d.d.disallowDuplicateFields = true
 }
 
+func (d *decodeState) newFieldError(msg, field string) error {
+	if len(d.strictFieldStack) > 0 {
+		return fmt.Errorf("%s %q", msg, strings.Join(d.strictFieldStack, "")+"."+field)
+	} else {
+		return fmt.Errorf("%s %q", msg, field)
+	}
+}
+
 // saveStrictError saves a strict decoding error,
 // for reporting at the end of the unmarshal if no other errors occurred.
 func (d *decodeState) saveStrictError(err error) {
```

Comment:
Do you think we should add a similar excessive-number-of-errors to to the pruning and conversion error accumulations I added?

Reply:
mm... we could... I don't feel as strongly about it there as in a low-level decoding library